src/pybind/mgr/smb/module.py

```diff
@@ -411,6 +411,45 @@ def show(
             return resources[0].to_simplified()
         return {"resources": [r.to_simplified() for r in resources]}
 
+    @cli.SMBCommand('share update cephfs qos', perm='rw')
+    def share_update_qos(
+        self,
+        cluster_id: str,
+        share_id: str,
+        read_iops_limit: Optional[int] = None,
+        write_iops_limit: Optional[int] = None,
+        read_bw_limit: Optional[int] = None,
+        write_bw_limit: Optional[int] = None,
+    ) -> results.Result:
+        """Update QoS settings for a CephFS share"""
+        try:
+            shares = self._handler.matching_resources(
+                [f'ceph.smb.share.{cluster_id}.{share_id}']
+            )
+            if not shares or not isinstance(shares[0], resources.Share):
+                raise ValueError(f"Share {cluster_id}/{share_id} not found")
+
+            share = shares[0]
+            if not share.cephfs:
+                raise ValueError("Share has no CephFS configuration")
+
+            updated_cephfs = share.cephfs.with_qos_updates(
+                read_iops_limit=read_iops_limit,
+                write_iops_limit=write_iops_limit,
+                read_bw_limit=read_bw_limit,
+                write_bw_limit=write_bw_limit,
+            )
+
+            share_data = share.to_simplified()
```

Comment:
Using `to_simplified` here just to reconstruct a new share object from the data seems awkward. See my comments elsewhere about helper methods and `dataclasses.replace`. If you're not sure what to do after reading my comments, let's talk.